src/cantools/database/can/message.py

```diff
@@ -750,36 +750,32 @@ def _assert_signal_values_valid(self,
                 if signal_value_num is None:
                     raise EncodeError(f'Invalid value specified for signal '
                                       f'"{signal.name}": "{signal_value}"')
-
                 continue
 
-            if signal.minimum is not None:
-                min_effective = signal.minimum
+            if scaling:
+                scaled_value = signal_value
+            else:
+                if signal.conversion.choices and signal_value in signal.conversion.choices:
+                    # skip range check if raw value exists in value table
+                    continue
 
-                # undo the scaling of the signal's minimum value if we
-                # are not supposed to scale the input value
-                if not scaling:
-                    min_effective = signal.conversion.numeric_scaled_to_raw(signal.minimum)
+                # scale signal to perform range check against minimum and maximum
+                scaled_value = signal.conversion.raw_to_scaled(
+                    raw_value=signal_value, decode_choices=False)  # type: ignore[assignment]
```

Comment:
possibly it is better to cast this to `float` instead of using `# type: ignore`:
```suggestion
                scaled_value = cast(float, signal.conversion.raw_to_scaled(
                    raw_value=signal_value, decode_choices=False))
```

Reply:
Done